src/lib/ui/thermostat/ClimateSettingStatus.tsx

```diff
@@ -59,40 +71,59 @@ function ClimateSettingIcon(props: {
 
 function Content(props: {
   mode: ClimateSetting['hvac_mode_setting']
-  coolingSetPoint: string
-  heatingSetPoint: string
+  coolingSetPoint: SetPoint
+  heatingSetPoint: SetPoint
   temperatureUnit: 'fahrenheit' | 'celsius'
 }): JSX.Element | null {
   const { mode, coolingSetPoint, heatingSetPoint, temperatureUnit } = props
   const hasCoolingSetPoint = coolingSetPoint !== undefined
   const hasHeatingSetPoint = heatingSetPoint !== undefined
-  const unit =
-    temperatureUnit === 'fahrenheit' ? t.degreeFahrenheit : t.degreeCelsius
 
   if (mode === 'cool' && hasCoolingSetPoint)
-    return <span>{`${coolingSetPoint}${unit}`}</span>
+    return (
+      <Temperature
+        fahrenheit={coolingSetPoint.fahrenheit}
+        celsius={coolingSetPoint.celsius}
+        unit={temperatureUnit}
+      />
+    )
 
   if (mode === 'heat' && hasHeatingSetPoint)
-    return <span>{`${heatingSetPoint}${unit}`}</span>
+    return (
+      <Temperature
+        fahrenheit={heatingSetPoint.fahrenheit}
+        celsius={heatingSetPoint.celsius}
+        unit={temperatureUnit}
+      />
+    )
 
   if (mode === 'heatcool' && hasHeatingSetPoint && hasCoolingSetPoint)
     return (
-      <span>{`${heatingSetPoint}${unit} - ${coolingSetPoint}${unit}`}</span>
+      <span>
+        <Temperature
+          fahrenheit={heatingSetPoint.fahrenheit}
+          celsius={heatingSetPoint.celsius}
+          unit={temperatureUnit}
+        />
+        {' - '}
+        <Temperature
+          fahrenheit={coolingSetPoint.fahrenheit}
+          celsius={coolingSetPoint.celsius}
+          unit={temperatureUnit}
+        />
+      </span>
     )
 
   if (mode === 'off') return <span>{t.off}</span>
 
   return null
 }
 
-const formatTemperatureValue = (value: number | undefined): string => {
-  if (value == null) return '0'
-  if (Number.isInteger(value)) return value.toFixed()
-  return value.toFixed(1)
+const formatTemperatureValue = (value: number | undefined): number => {
```

Comment:
Was there a reason to change this? Keeping one decimal place was intentional. Maybe we should only show decimal for celsius?